src/util/newconfig/ConfigDescription.hpp

```diff
@@ -61,26 +72,78 @@ struct ClioConfigDescription {
         return itr->value;
     }
 
+    /**
+     * @brief Generate markdown file of all the clio config descriptions
+     *
+     * @param path The path location to generate the Config-description file
+     * @return An Error if generating markdown fails, otherwise nothing
+     */
+    [[nodiscard]] static std::expected<void, Error>
+    getMarkdown(std::string_view path)
```

Comment:
By the name of this function I'd expect it to return the MD as a string or an error if generation failed. It does more than the name suggests - creates a file, writes to file, prints to standard output. I think it would be cleaner if the handling of the generated MD was outside of this function. Consider splitting to smaller functions.